flink-table-store-core/src/main/java/org/apache/flink/table/store/file/FileStoreImpl.java

```diff
@@ -232,12 +235,29 @@ public static FileStoreImpl createWithPrimaryKey(
                 mergeFunction = new DeduplicateMergeFunction();
                 break;
             case PARTIAL_UPDATE:
-                List<LogicalType> fieldTypes = rowType.getChildren();
-                RowData.FieldGetter[] fieldGetters = new RowData.FieldGetter[fieldTypes.size()];
-                for (int i = 0; i < fieldTypes.size(); i++) {
-                    fieldGetters[i] = RowData.createFieldGetter(fieldTypes.get(i), i);
+                mergeFunction = new PartialUpdateMergeFunction(rowType);
+                break;
+            case AGGREGATION:
+                Map<String, String> rightConfMap =
+                        options.getFilterConf(e -> e.getKey().endsWith(".aggregate-function"));
+                Set<String> aggregateColumnNames =
+                        rightConfMap.keySet().stream()
+                                .distinct()
+                                .flatMap(s -> Stream.of(s.split(".aggregate-function")[0]))
+                                .collect(Collectors.toSet());
+                switch (AggregateFunctionFactory.getAggregationKind(rightConfMap.values())) {
+                    case Sum:
+                        mergeFunction =
+                                new SumAggregateMergeFunction(
+                                        primaryKeyType, rowType, aggregateColumnNames);
+                        break;
+                    case Avg:
+                    case Max:
+                    case Min:
+                    default:
+                        throw new UnsupportedOperationException(
+                                "merge-function values un supposed");
```

Comment:
Extract this to a separated factory class. We'd like each method to be simple. Maybe the outer `switch` statement should also be extracted?

If you're extracting both `switch` statements I would suggest creating 2 factory classes. Each class should complete one specific task.